planner/core/logical_plan_builder.go

```diff
@@ -4515,6 +4468,54 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 		return nil, ErrPartitionClauseOnNonpartitioned
 	}
 
+	possiblePaths, err := getPossibleAccessPaths(b.ctx, b.TableHints(), tn.IndexHints, tbl, dbName, tblName, b.isForUpdateRead, b.optFlag&flagPartitionProcessor > 0)
```

Comment:
Could you explain why move the codes from L4418 to here?

Reply:
The code above `if tableInfo.GetPartitionInfo() != nil` affects flagPartitionProcessor.
`getPossibleAccessPaths` method has a new parameter，which depends on flagPartitionProcessor.